zulip_bots/zulip_bots/test_run_py.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import
+
+#!/usr/bin/env python
+
+import importlib
+import os
+import zulip_bots.run
+import six
+import unittest
+import zulip
+
+from importlib import import_module
+from unittest import TestCase
+
+if six.PY2:
+    import mock
+    from mock import patch
+else:
+    from unittest import mock
+    from unittest.mock import patch
+
+class TestDefaultArguments(TestCase):
+
+    our_dir = os.path.dirname(__file__)
+    path_to_bot = os.path.abspath(os.path.join(our_dir, 'bots/giphy/giphy.py'))
+
+    @patch('sys.argv', ['zulip-run-bot', 'giphy', '--config-file', '/foo/bar/baz.conf'])
+    @patch('zulip_bots.run.run_message_handler_for_bot')
+    def test_argument_parsing_with_bot_name(self, mock_run_message_handler_for_bot):
+        zulip_bots.run.main()
+        mock_run_message_handler_for_bot.assert_called_with(bot_name='giphy',
+                                                            config_file='/foo/bar/baz.conf',
+                                                            lib_module=mock.ANY,
+                                                            quiet=False)
+
+    @patch('sys.argv', ['zulip-run-bot', '--path-to-bot', path_to_bot, '--config-file', '/foo/bar/baz.conf'])
+    @patch('zulip_bots.run.run_message_handler_for_bot')
+    def test_argument_parsing_with_bot_path(self, mock_run_message_handler_for_bot):
+        zulip_bots.run.main()
+        mock_run_message_handler_for_bot.assert_called_with(bot_name=None,  # TODO: This should be equal to the bot's name
+                                                            config_file='/foo/bar/baz.conf',
+                                                            lib_module=mock.ANY,
+                                                            quiet=False)
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
I ran these tests on my machine and they passed (yay)! One thing that concerns me is the output of the tests; it isn't very informative:
```
..
----------------------------------------------------------------------
Ran 2 tests in 0.004s

OK
```

I think we should increase the verbosity of the test runner to display the names of the passing (or failing) tests (a convention we seem to follow a lot in our tests). See [here](https://github.com/zulip/python-zulip-api/blob/master/tools/test-bots#L95). On a similar note, I don't see these tests being run anywhere as part of Travis or anything, so I think we should do that too! :)

Perhaps another thing we need to discuss (not as a part of this PR) is whether we need to "discover" these tests in `tools/test-bots` and run it from a master "test all" script for this package. :)

Reply:
Travis seems to be running the tests (see [here](https://travis-ci.org/zulip/python-zulip-api/jobs/272814581#L555), lines 542-543), as these tests are already discovered by `tools/test-bots`. The script running all test suites is `test-main` (as opposed to `test-all`, since it doesn't run the linter). We made those changes as part of a refactoring of our test suites. 

Regarding the verbosity on single executions of this test suite, I oriented myself by  similar tests in this repo, e.g. `test_server.py`. We might want to make another PR though, that'll update all test files to be verbose.